### PR TITLE
Feature/fetch content

### PR DESCRIPTION
**Pull request by Leon0402**
Hi,

as discussed in #348 my implementation to use fetch content instead. I've also added instructions on how to use the lib and two examples (one for fetch content and one for find package).

I also made a little bit more fetch friendly for example by including CTest (which will automatically provide a BUILD_TESTING cache variable) and automatically deactive test, if it's not the top level project.

The main.cpp is empty in both though, do you have a very short code using any of the methods from the lib? Just to verify that everything is working. I browsed in the tests, but they seemed a little bit too complicated for this simple example. It doesn't have to be a useful example.

### CODE REVIEW

**Comment by rbock** [comment]
Thanks for the pull request! I will need a bit of time to do a more thorough review.

For main.cpp
`test_serializer/Insert.cpp` might be a good inspiration? 

`compare(__LINE__, insert_into(bar).default_values(), "INSERT INTO tab_bar DEFAULT VALUES");` could be a good one-liner (plus includes).

When quickly scrolling through, I noticed that you set the C++ standard to C++14? Ideally, it should still work with C++11.

**Comment by Leon0402** [comment]
No worries, take your time. I will be also happy to explain you some thing, if something is unclear. Btw. where is the travis build, shouln't it show up here? 

Thanks for the example, I will have a look. I also wondered if I should include the two examples in the pipeline? As kind of an deployment test? I'm not a travis expert, but I think this would be possible and helpful. 

> When quickly scrolling through, I noticed that you set the C++ standard to C++14? Ideally, it should still work with C++11.

This is only the C++ Version of the examples, it was just randomly chosen :) But I agree, it would be best to set it to the minimum version possible for testing purposes. I will change it

**Comment by Leon0402** [comment]
> For main.cpp
> `test_serializer/Insert.cpp` might be a good inspiration?
> 
> `compare(__LINE__, insert_into(bar).default_values(), "INSERT INTO tab_bar DEFAULT VALUES");` could be a good one-liner (plus includes).

The problem is that this needs compare.h, which needs MockDb.h ... I don't wont to depend on these files, because people just just use that sample as a quick test to see if everything is working (so use it outside of this repository). 
Anything else, which just depends on the official headers?

**Comment by rbock** [comment]
OK, so how about this as a one-liner?

```
select(sqlpp::value(false).as(sqlpp::alias::a)));
```

Does not do anything, but if this compiles, we should be in good shape.

We could serialize **without** MockDb, if you want some runtime test, too.

**Comment by rbock** [comment]
Re Travis: Thanks for noticing. They are here: https://travis-ci.org/github/rbock/sqlpp11/pull_requests and they failed. Strange that they are not shown.

**Comment by Leon0402** [comment]
I think this should be ready except the fact that the url in fetchConent is still pointing to my repo. (So I will leave it a draft). When you approve the pull request I will change that to your repo and the development branch.

I tested both the find_package as well as the fetchConent approach on my Linux machine.

File: dependencies/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2013-2016, Roland Bock
+# Copyright (c) 2016 Christian Dávid
```
Comment (rbock) [inline dependencies/CMakeLists.txt:2]: I suppose you should add your name and 2020 here instead?

File: dependencies/hinnant_date/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2013-2016, Roland Bock
+# Copyright (c) 2016 Christian Dávid
```
Comment (rbock) [inline dependencies/hinnant_date/CMakeLists.txt:2]: Same here.

**Comment by rbock** [comment]
Looks great!

In addition to the two comments: Please add copyright and license to the new example files, too.

**Comment by Leon0402** [comment]
Okay all done :)

**Comment by rbock** [comment]
Great! Assuming that the checks pass (strange that travis still does not show up), I'll merge during the weekend.

**Comment by Leon0402** [comment]
> Great! Assuming that the checks pass (strange that travis still does not show up), I'll merge during the weekend.

I saw that issue on other repos as well, a quick search found this
https://travis-ci.community/t/travis-ci-build-no-longer-showing-as-check-on-github-pull-request-page/6955/10

Perhaps you have to renew something?

**Comment by rbock** [comment]
Thanks for the hint. Apparently, I needed to sign up to beta to migrate some of the repositories.

**Comment by rbock** [comment]
I updated the readme to point to travis-ci.com now. Could you rebase, please? We could see if the travis build shows up again :-)

**Comment by Leon0402** [comment]
Seems to work again :)

**Comment by rbock** [comment]
Great! I would squash and merge using these comments, OK?

* Add date as a dependency, make fetch friendly
* Add example
* Add alias
* Update readme
* Add find_package example
* Update travis

**Comment by Leon0402** [comment]
Sure, go ahead! :+1:

**Comment by rbock** [comment]
Done.
I'll let this stew for a couple of days before tagging a new release.

**Pull request merged**